### PR TITLE
[CARBONDATA-3639][CARBONDATA-3638] Fix global sort exception in load from CSV flow with binary non-sort columns

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Problem:
Global sort throws exception in load from CSV flow with binary non-sort columns.
Exception is attached in JIRA and test case is added in PR.

Cause: For global sort flow, we make csvRDD from file. But again this String RDD is converted to new string scala objects, here binary was exceeding 32000 length. Hence the exception

Solution: For CSV load flow, avoid this new String object conversion as it is already string object.
This can also handle [CARBONDATA-3638]

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? NA
 
 - [ ] Any backward compatibility impacted? NA
 
 - [ ] Document update required? NA

 - [ ] Testing done
        Added UT.       

 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. Done.

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1338/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/1325/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1319/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala [hunk 1]
```diff
@@ -88,11 +90,24 @@ object DataLoadProcessBuilderOnSpark {
...
     // 1. Input
-    val inputRDD = originRDD
-      .mapPartitions(rows => DataLoadProcessorStepOnSpark.toRDDIterator(rows, modelBroadcast))
-      .mapPartitionsWithIndex { case (index, rows) =>
-        DataLoadProcessorStepOnSpark.inputFunc(rows, index, modelBroadcast, inputStepRowCounter)
+    val inputRDD = if (isLoadFromCSV) {
+      // No need of wrap with NewRDDIterator, which converts object to string,
+      // as it is already a string.
+      // So, this will avoid new object creation in case of CSV global sort load for each row
+      originRDD.mapPartitionsWithIndex { case (index, rows) => DataLoadProcessorStepOnSpark
```
Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala]: Move DataLoadProcessorStepOnSpark to next line

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessorStepOnSpark.scala [hunk 2]
```diff
@@ -97,6 +97,36 @@ object DataLoadProcessorStepOnSpark {
...
+      wrapException(e, model)
+    }
+
+    new Iterator[CarbonRow] {
+      override def hasNext: Boolean = rows.hasNext
+
+      override def next(): CarbonRow = {
+        var row : CarbonRow = null
+        val rawRow = rows.next().values.asInstanceOf[Array[Object]]
+        if(isRawDataRequired) {
```
Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessorStepOnSpark.scala]:
```suggestion
        if (isRawDataRequired) {
```

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/binary/TestBinaryDataType.scala [hunk 3]
```diff
@@ -112,6 +112,48 @@ class TestBinaryDataType extends QueryTest with BeforeAndAfterAll {
...
+        sql("DROP TABLE IF EXISTS binaryTable")
+        sql(
+            s"""
+               | CREATE TABLE IF NOT EXISTS binaryTable (
+               |    id int,
+               |    label boolean,
+               |    name string,
+               |    binaryField binary,
+               |    autoLabel boolean)
+               | STORED BY 'carbondata'
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/binary/TestBinaryDataType.scala]: use STORED AS CARBONDATA

Comment (ajantha-bhat) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/binary/TestBinaryDataType.scala]: OK. all other testcases in the file was using 'stored by'

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessorStepOnSpark.scala (hunk 2 above)
Comment (ajantha-bhat) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessorStepOnSpark.scala]: done

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala (hunk 1 above)
Comment (ajantha-bhat) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala]: done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.1.0, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.1/1340/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.2.1, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.2/1350/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1364/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/1450/

**Pull request closed without merging**